v2/failure-injected-spanner-service/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerServiceFactoryImpl.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.beam.sdk.io.gcp.spanner;
+
+import com.google.cloud.teleport.v2.spanner.service.SpannerService;
+
+public class SpannerServiceFactoryImpl {
+  public static SpannerConfig createSpannerService(SpannerConfig spannerConfig, String parameter) {
+    return spannerConfig.withServiceFactory(new SpannerService(parameter));
```

Comment:
I think you should check if `spannerConfig` already does not have a service factory instantiated in it?

Reply:
Done.